<launch>
  <!-- SLAM algorithm (dxslam / orbslam2) -->
  <arg name="slam" default="dxslam"/>
  <!-- default prefix of all the input topics -->
  <arg name="camera" default="camera"/>
  <!-- topic of the color images -->
  <arg name="image_topic" default="$(arg camera)/color/image_raw"/>
  <!-- topic of the depth images -->
  <arg name="depth_topic" default="$(arg camera)/aligned_depth_to_color/image_raw"/>
  <!-- topic of the features of the color images -->
  <arg name="feature_topic" default="$(arg camera)/color/features"/>
  <!-- parent frame for publishing poses to /tf -->
  <arg name="pub_tf_parent_frame" default="map"/>
  <!-- child frame for publishing poses to /tf; usually "base_link" or "camera_link";
       leave blank to use the frame specified in the color image messages -->
  <arg name="pub_tf_child_frame" default=""/>
  <!-- config file, either filename under the config folder or an absolute path -->
  <arg name="config" default="realsense_d435.yaml"/>
  <!-- whether to launch with gdb for debugging -->
  <arg name="gdb" default="false"/>
  <arg name="launch_prefix" default="gdb -ex run -args" if="$(arg gdb)"/>
  <arg name="launch_prefix" default="" unless="$(arg gdb)"/>

  <!-- add a frame /vslam_origin, denoting the initial pose of the camera (optical frame) -->
  <node name="map_vslam_tf" pkg="tf" type="static_transform_publisher" args="0 0 0 -1.5708 0 -1.5708 $(arg pub_tf_parent_frame) /vslam_origin 100" />

  <node pkg="dxslam_ros" type="$(arg slam)" name="$(arg slam)" output="screen" launch-prefix="$(arg launch_prefix)">
    <param name="config" value="$(arg config)"/>
    <param name="reference_frame" value="vslam_origin"/>
    <param name="pub_tf_parent_frame" value="vslam_origin"/>
    <param name="pub_tf_child_frame" value="$(arg pub_tf_child_frame)"/>
    <param name="image_topic" value="$(arg image_topic)"/>
    <param name="depth_topic" value="$(arg depth_topic)"/>
    <param name="feature_topic" value="$(arg feature_topic)"/>
  </node>

</launch>
